lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,7 +154,10 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
-        evt_time = time()
+        if user.is_anonymous():
+            # Anonymous users cannot be persisted to the database, so let's just use
+            # what we have.
+            return
```

Comment:
On the anonymous student handling: Great. :-)

On the Datadog metrics removal -- why are we doing this? We have dashboards that track this stuff in Datadog, and had an entire run up where we thought we might need to switch our student state backing store to address load issues. I don't understand the motivation for removing it.